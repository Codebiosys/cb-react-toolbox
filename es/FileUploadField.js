'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactBootstrap = require('react-bootstrap');

require('./FileUploadField.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FAKE_PATH_PATTERN = /C:\\fakepath\\/i;

var FileUploadField = function (_React$Component) {
  (0, _inherits3.default)(FileUploadField, _React$Component);

  function FileUploadField(props) {
    (0, _classCallCheck3.default)(this, FileUploadField);

    var _this = (0, _possibleConstructorReturn3.default)(this, (FileUploadField.__proto__ || (0, _getPrototypeOf2.default)(FileUploadField)).call(this, props));

    _this.componentWillReceiveProps = function (_ref) {
      var value = _ref.value;

      // Reset the filename when props change.
      if (!value) {
        _this.setState((0, _extends3.default)({}, _this.state, {
          filename: ''
        }));
        _this.formInput.value = '';
      }
    };

    _this.handleBrowse = function () {
      _this.formInput.click();
    };

    _this.handleFileWasUpdated = function () {
      var filename = _this.formInput.value.replace(FAKE_PATH_PATTERN, '');
      _this.setState({ filename: filename });

      // Alert parent components.
      _this.props.onChange(_this.formInput.files[0]);
    };

    _this.render = function () {
      var placeholder = _this.props.placeholder;
      var filename = _this.state.filename;

      return _react2.default.createElement(
        'div',
        { className: 'form-group' },
        _react2.default.createElement('input', {
          type: 'file',
          name: 'file',
          className: 'file',
          onChange: _this.handleFileWasUpdated,
          ref: function ref(e) {
            _this.formInput = e;
          }
        }),
        _react2.default.createElement(
          'div',
          { className: 'input-group col-xs-12' },
          _react2.default.createElement(
            'span',
            { className: 'input-group-addon' },
            _react2.default.createElement('i', { className: 'glyphicon glyphicon-file' })
          ),
          _react2.default.createElement(
            'span',
            {
              type: 'text',
              className: 'form-control text-center input-lg file-input-text-input',
              disabled: true,
              readOnly: true
            },
            filename || _react2.default.createElement(
              'span',
              { className: 'text-muted' },
              placeholder
            )
          ),
          _react2.default.createElement(
            'span',
            { className: 'input-group-btn' },
            _react2.default.createElement(
              _reactBootstrap.Button,
              { className: 'browse btn btn-default input-lg', onClick: _this.handleBrowse },
              _react2.default.createElement('i', { className: 'glyphicon glyphicon-search' }),
              '\xA0 Browse'
            )
          )
        )
      );
    };

    _this.state = {
      filename: props.value.name || props.value
    };
    return _this;
  }

  return FileUploadField;
}(_react2.default.Component);

FileUploadField.propTypes = {
  placeholder: _propTypes2.default.string,
  onChange: _propTypes2.default.func,
  value: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.object])
};
FileUploadField.defaultProps = {
  value: '',
  placeholder: '',
  onChange: function onChange() {}
};
exports.default = FileUploadField;