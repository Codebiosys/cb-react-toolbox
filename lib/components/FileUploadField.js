'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactBootstrap = require('react-bootstrap');

require('./FileUploadField.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var FAKE_PATH_PATTERN = /C:\\fakepath\\/i;

var FileUploadField = function (_React$Component) {
  _inherits(FileUploadField, _React$Component);

  function FileUploadField(props) {
    _classCallCheck(this, FileUploadField);

    var _this = _possibleConstructorReturn(this, (FileUploadField.__proto__ || Object.getPrototypeOf(FileUploadField)).call(this, props));

    _this.componentWillReceiveProps = function (_ref) {
      var value = _ref.value;

      // Reset the filename when props change.
      if (!value) {
        _this.setState(_extends({}, _this.state, {
          filename: ''
        }));
        _this.formInput.value = '';
      }
    };

    _this.handleBrowse = function () {
      _this.formInput.click();
    };

    _this.handleFileWasUpdated = function () {
      var filename = _this.formInput.value.replace(FAKE_PATH_PATTERN, '');
      _this.setState({ filename: filename });

      // Alert parent components.
      _this.props.onChange(_this.formInput.files[0]);
    };

    _this.render = function () {
      var placeholder = _this.props.placeholder;
      var filename = _this.state.filename;

      return _react2.default.createElement(
        'div',
        { className: 'form-group' },
        _react2.default.createElement('input', {
          type: 'file',
          name: 'file',
          className: 'file',
          onChange: _this.handleFileWasUpdated,
          ref: function ref(e) {
            _this.formInput = e;
          }
        }),
        _react2.default.createElement(
          'div',
          { className: 'input-group col-xs-12' },
          _react2.default.createElement(
            'span',
            { className: 'input-group-addon' },
            _react2.default.createElement('i', { className: 'glyphicon glyphicon-file' })
          ),
          _react2.default.createElement(
            'span',
            {
              type: 'text',
              className: 'form-control text-center input-lg file-input-text-input',
              disabled: true,
              readOnly: true
            },
            filename || _react2.default.createElement(
              'span',
              { className: 'text-muted' },
              placeholder
            )
          ),
          _react2.default.createElement(
            'span',
            { className: 'input-group-btn' },
            _react2.default.createElement(
              _reactBootstrap.Button,
              { className: 'browse btn btn-default input-lg', onClick: _this.handleBrowse },
              _react2.default.createElement('i', { className: 'glyphicon glyphicon-search' }),
              '\xA0 Browse'
            )
          )
        )
      );
    };

    _this.state = {
      filename: props.value.name || props.value
    };
    return _this;
  }

  return FileUploadField;
}(_react2.default.Component);

FileUploadField.defaultProps = {
  value: '',
  placeholder: '',
  onChange: function onChange() {}
};
exports.default = FileUploadField;