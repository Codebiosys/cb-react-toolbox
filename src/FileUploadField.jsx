import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

import './FileUploadField.css';

const FAKE_PATH_PATTERN = /C:\\fakepath\\/i;

class FileUploadField extends React.Component {
  static propTypes = {
    placeholder: PropTypes.string,
    onChange: PropTypes.func,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  }

  static defaultProps = {
    value: '',
    placeholder: '',
    onChange: () => {},
  }

  constructor(props) {
    super(props);

    this.state = {
      filename: props.value.name || props.value,
    };
  }

  componentWillReceiveProps = ({ value }) => {
    // Reset the filename when props change.
    if (!value) {
      this.setState({
        ...this.state,
        filename: '',
      });
      this.formInput.value = '';
    }
  };

  handleBrowse = () => {
    this.formInput.click();
  };

  handleFileWasUpdated = () => {
    const filename = this.formInput.value.replace(FAKE_PATH_PATTERN, '');
    this.setState({ filename });

    // Alert parent components.
    this.props.onChange(this.formInput.files[0]);
  };

  render = () => {
    const { placeholder } = this.props;
    const { filename } = this.state;
    return (
      <div className="form-group">
        <input
          type="file"
          name="file"
          className="file"
          onChange={this.handleFileWasUpdated}
          ref={(e) => { this.formInput = e; }}
        />
        <div className="input-group col-xs-12">
          <span className="input-group-addon">
            <i className="glyphicon glyphicon-file" />
          </span>
          <span
            type="text"
            className="form-control text-center input-lg file-input-text-input"
            disabled
            readOnly
          >
            {filename || <span className="text-muted">{placeholder}</span>}
          </span>
          <span className="input-group-btn">
            <Button className="browse btn btn-default input-lg" onClick={this.handleBrowse}>
              <i className="glyphicon glyphicon-search" />&nbsp;
              Browse
            </Button>
          </span>
        </div>
      </div>
    );
  }
}

export default FileUploadField;
