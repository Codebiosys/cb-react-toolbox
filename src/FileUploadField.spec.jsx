import React from 'react';
import { shallow } from 'enzyme';
import FileUploadField from './FileUploadField';

const componentProps = {
};

describe('The File Upload Field', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <FileUploadField {...componentProps} />,
    );
  });
  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
