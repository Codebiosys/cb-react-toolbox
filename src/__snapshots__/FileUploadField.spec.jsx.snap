// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The File Upload Field renders 1`] = `
<div
  className="form-group"
>
  <input
    className="file"
    name="file"
    onChange={[Function]}
    type="file"
  />
  <div
    className="input-group col-xs-12"
  >
    <span
      className="input-group-addon"
    >
      <i
        className="glyphicon glyphicon-file"
      />
    </span>
    <span
      className="form-control text-center input-lg file-input-text-input"
      disabled={true}
      readOnly={true}
      type="text"
    >
      <span
        className="text-muted"
      />
    </span>
    <span
      className="input-group-btn"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="browse btn btn-default input-lg"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="glyphicon glyphicon-search"
        />
        Â  Browse
      </Button>
    </span>
  </div>
</div>
`;
